src/ota_imxrt.c

```diff
@@ -194,8 +194,8 @@ MG_IRAM static bool flash_page_start(volatile uint32_t *dst) {
 // Note: the get_config function below works both for RT1020 and 1060
 #if MG_OTA == MG_OTA_RT1020
 MG_IRAM static int flexspi_nor_get_config(
-    struct mg_flexspi_nor_config *config) {
-  struct mg_flexspi_nor_config default_config = {
+  struct mg_flexspi_nor_config *config) {
+  static struct mg_flexspi_nor_config default_config = {
```

Comment:
Does `static` make a difference ? If it is dynamic, the compiler should generate a bunch of code to fill the struct in the stack, then memory released. If it is static, the compiler will probably init this function at startup and memory be used forever.

Does our nice compiler use stdlib calls to fill this struct in the stack ?

Then I think our copy doesn't make much sense, we should probably declare this as module static (not function static) and name it and/or change the code to access this data. (With a nice comment remembering us that this must reside in RAM)

Reply:
If we do not use `static`, then the structure will be allocated on the stack, the compiler uses memset to zero it, and then it initializes it with the corresponding values. If we use 'static', memset is no longer used, the variable is probably initialized as global and the code just loads its address.

Here is the generated code when static is not used. Note that it reserves space for the variable on stack and then clears it with memset and then proceeds to fill it with data.
```
00000e1c <flexspi_nor_get_config>:
     e1c:	b510      	push	{r4, lr}
     e1e:	f5ad 7d02 	sub.w	sp, sp, #520	; 0x208
     e22:	f44f 7200 	mov.w	r2, #512	; 0x200
     e26:	2100      	movs	r1, #0
     e28:	4604      	mov	r4, r0
     e2a:	ab02      	add	r3, sp, #8
     e2c:	4618      	mov	r0, r3
     e2e:	f000 f9b3 	bl	1198 <__memset_veneer>
     e32:	4924      	ldr	r1, [pc, #144]	; (ec4 <flexspi_nor_get_config+0xa8>)
     e34:	4a24      	ldr	r2, [pc, #144]	; (ec8 <flexspi_nor_get_config+0xac>)
..........
```

When static is used, the code is this:
```
0000101c <flexspi_nor_get_config>:
    101c:	b082      	sub	sp, #8
    101e:	2300      	movs	r3, #0
    1020:	4a08      	ldr	r2, [pc, #32]	; (1044 <flexspi_nor_get_config+0x28>)
    1022:	9301      	str	r3, [sp, #4]
    1024:	9b01      	ldr	r3, [sp, #4]
    1026:	f5b3 7f00 	cmp.w	r3, #512	; 0x200
    102a:	d302      	bcc.n	1032 <flexspi_nor_get_config+0x16>
    102c:	2000      	movs	r0, #0
    102e:	b002      	add	sp, #8
    1030:	4770      	bx	lr
    1032:	9901      	ldr	r1, [sp, #4]
    1034:	9b01      	ldr	r3, [sp, #4]
    1036:	5c51      	ldrb	r1, [r2, r1]
    1038:	54c1      	strb	r1, [r0, r3]
    103a:	9b01      	ldr	r3, [sp, #4]
    103c:	3301      	adds	r3, #1
    103e:	9301      	str	r3, [sp, #4]
    1040:	e7f0      	b.n	1024 <flexspi_nor_get_config+0x8>
    1042:	bf00      	nop
    1044:	00000008 	.word	0x00000008
```

Regarding the copy,

```*config = default_config;``` always results in the compiler using memcpy, like you said.

So I am using a byte-by-byte copy in order to avoid the compiler inventing another optimization with 'memcpy'